services/blockchain-connector/shared/sdk/blocksUtils.js

```diff
@@ -80,28 +76,46 @@ const loadConfig = async () => {
 	}
 };
 
-const downloadGenesisBlock = async () => {
+// eslint-disable-next-line consistent-return
+const downloadAndValidateGenesisBlock = async (retries = 2) => {
 	const directoryPath = path.dirname(genesisBlockFilePath);
-	if (!(await exists(directoryPath))) await mkdir(directoryPath, { recursive: true });
+	const genesisFileName = genesisBlockUrl.substring(genesisBlockUrl.lastIndexOf('/') + 1);
+	const genesisFilePath = `${directoryPath}/${genesisFileName}`;
+	const checksumFilePath = `${genesisFilePath}.SHA256`;
 
-	logger.info(`Downloading genesis block to the filesystem from: ${genesisBlockUrl}`);
+	do {
+		/* eslint-disable no-await-in-loop */
+		// Delete all previous files including the containing directory
+		await deleteFileRecursive(directoryPath);
```

Comment:
Please delete the contents only when the checksum validation fails.
```suggestion
```

Reply:
Applied